iroh-net/src/magicsock/endpoint.rs

```diff
@@ -153,8 +150,9 @@ impl Endpoint {
 
     /// Returns info about this endpoint
     pub fn info(&self) -> EndpointInfo {
-        let (conn_type, latency) = if self.is_best_addr_valid(Instant::now()) {
-            let addr_info = self.best_addr.as_ref().expect("checked");
+        // TODO: I think we should return direct here for outdated.
```

Comment:
to me it makes sense what you say. When it's outdated we return both this address and the derp and metrics are only updated when the address is cleared. So even there, we are counting an outdated conn as a a direct one

Reply:
I changed it so that we return `Direct` if best_addr is set but expired.

Actually the more proper fix might be to return something like `ConnectionType::Both` because if the `best_addr` is set but expired, we send over both Derp and the Udp address. But not sure how important this is to report to the outside.

Reply:
In the most recent commit I changed this to return `ConnectionType::Mixed`  in that case. Also put it all into a single `match` statement for clarity.